src/main/java/org/folio/circulation/services/ItemsInTransitReportService.java

```diff
@@ -79,9 +84,15 @@
   }
 
   private CompletableFuture<Result<ItemsInTransitReportContext>> fetchInstances(
-    Result<ItemsInTransitReportContext> context) {
+    ItemsInTransitReportContext context) {
 
-    return completedFuture(context);
+    var mapper = new InstanceMapper();
+    FindWithMultipleCqlIndexValues<Instance> fetcher
+      = findWithMultipleCqlIndexValues(instancesStorageClient, "instances", mapper::toDomain);
+
+    return fetcher.findByIds(context.getItems().keySet())
```

Comment:
So, you are looking for instances by their ids if I'm not mistaken of course but sent to this method itemIds (context.getItems().keySet()), right ?

Reply:
right, fixed. Thanks